src/nvim/liveupdate.c

```diff
@@ -0,0 +1,195 @@
+#include "nvim/liveupdate.h"
+#include "nvim/memline.h"
+#include "nvim/api/private/helpers.h"
+#include "nvim/msgpack_rpc/channel.h"
+
+// Register a channel. Return True if the channel was added, or already added.
+// Return False if the channel couldn't be added because the buffer is
+// unloaded.
+bool liveupdate_register(buf_T *buf, uint64_t channel_id)
+{
+  // must fail if the buffer isn't loaded
+  if (buf->b_ml.ml_mfp == NULL) {
+    return false;
+  }
+
+  // count how many channels are currently watching the buffer
+  size_t size = kv_size(buf->liveupdate_channels);
+  if (size) {
+    for (size_t i = 0; i < size; i++) {
+      if (kv_A(buf->liveupdate_channels, i) == channel_id) {
+        // buffer is already registered ... nothing to do
+        return true;
+      }
+    }
+  }
+
+  // append the channelid to the list
+  kv_push(buf->liveupdate_channels, channel_id);
+
+  // send through the full channel contents now
+  Array linedata = ARRAY_DICT_INIT;
```

Comment:
I would very much like a flag to not include (or set to an empty array) the linedata in LiveUpdateStart (ActualVim already knows the entire buffer contents, and I'm just using neovim to track changes to the buffer. Not sending the whole buffer extra times is the entire reason I want a diff system.)

Reply:
I've mentally tagged this feature request as "pretty much everyone is asking for this". :-) I think it won't be hard to implement, I just need to find some time in the near future to get it done.